Mscorlib failing to build in some tests

Getting the following in http://dotnet-ci.cloudapp.net/job/dotnet_coreclr/job/master/job/x64_release_windows_nt_pri1_prtest/1928/

error CS1719: Error opening Win32 resource file 'D:\j\workspace\x86_ryujit_ch---7ecedcb1\bin\obj/\Windows_NT.x86.Checked\/Native.res' -- 'The process cannot access the file 'D:\j\workspace\x86_ryujit_ch---7ecedcb1\bin\obj\Windows_NT.x86.Checked\Native.res' because it is being used by another process.' [D:\j\workspace\x86_ryujit_ch---7ecedcb1\src\mscorlib\System.Private.CoreLib.csproj]
Indication 1 

This is usually an indication that certain msbuild properties are not being cleared/set properly in Project Reference dependencies, and a project is erroneously being built twice.

/cc @RussKeldorph @weshaggard @gkhanna79

--- a/src/mscorlib/System.Private.CoreLib.csproj
+++ b/src/mscorlib/System.Private.CoreLib.csproj
@@ -192,6 +192,12 @@
   <Import Project="$(MSBuildThisFileDirectory)Tools\Versioning\GenerateVersionInfo.targets"/>
   <!-- Override versioning targets -->
   <Import Condition="Exists('$(ToolsDir)versioning.targets')" Project="$(ToolsDir)versioning.targets" />
+
+  <PropertyGroup>
+    <!-- Use a different nativeresource file to avoid conflicts with mscorlib-->
+    <Win32Resource Condition="'$(GenerateNativeVersionInfo)'=='true'">$(IntermediateOutputPath)\System.Private.CoreLib.res</Win32Resource>
+  </PropertyGroup>
+
   <Import Project="GenerateSplitStringResources.targets"/>
   <Import Project="GenerateCompilerResponseFile.targets"/>
   <Import Project="$(PostProcessingToolsPath)" />

